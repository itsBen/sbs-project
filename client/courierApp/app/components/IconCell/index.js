import React from 'react';
import { Cell } from 'react-native-tableview-simple';
import Icon from 'react-native-vector-icons/MaterialIcons';

export default ({ iconName, ...props }) => (
  <Cell
    accessory="DisclosureIndicator"
    contentContainerStyle={{ paddingVertical: 10 }}
    cellImageView={
      <Icon
        name={iconName}
        size={22}
        color="rgb(153,153,153)"
        style={{ marginRight: 15 }}
      />
    }
    {...props}
  />
);
