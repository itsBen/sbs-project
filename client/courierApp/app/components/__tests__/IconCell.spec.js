/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import IconCell from '../IconCell';

it('renders correctly', () => {
  const tree = renderer.create(<IconCell iconName="location-on" />).toJSON();
  expect(tree).toMatchSnapshot();
});
