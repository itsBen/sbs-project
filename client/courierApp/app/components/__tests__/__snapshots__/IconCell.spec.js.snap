// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "minHeight": 44,
          "paddingLeft": 15,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#FFF",
        },
        Object {
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(153,153,153)",
            "fontSize": 22,
          },
          Object {
            "marginRight": 15,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      îƒˆ
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 16,
              "letterSpacing": -0.32,
            },
            Object {
              "color": "#000",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#c7c7cc",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "height": 10,
            "marginLeft": 7,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
